var/spack/repos/builtin/packages/bcl2fastq2/package.py

```diff
@@ -91,8 +94,11 @@ def wrap():
                 else:
                     tty.msg("Unpacking bcl2fastq2 tarball")
                     tty.msg("cwd sez: {0}".format(os.getcwd()))
-                    tarball = 'bcl2fastq2-v{0}.tar.gz'.format(
-                        self.version.dotted)
+                    ver = '{0}'.format(self.version.dotted)
+                    if ver == '2.20.0.422':
+                        tarball = 'bcl2fastq2-v{0}-Source.tar.gz'.format(ver)
+                    else:
+                        tarball = 'bcl2fastq2-v{0}.tar.gz'.format(ver)
```

Comment:
You could just replace this with a glob that doesn't care about version number of "Source". There will only be a single tarball inside.